apollo-router/src/plugins/telemetry/tracing/datadog.rs

```diff
@@ -27,6 +27,7 @@ lazy_static! {
     static ref SPAN_RESOURCE_NAME_ATTRIBUTE_MAPPING: HashMap<&'static str, &'static str> = {
         let mut map = HashMap::new();
         map.insert("request", "http.route");
+        map.insert("router", "http.route");
```

Comment:
I think based on [this latest comment](https://github.com/apollographql/router/issues/5282#issuecomment-2168849672), we have to `router` as the HTTP and Service entry span. I think this change could go in.